client/core/trade.go

```diff
@@ -1612,90 +1630,107 @@ func (c *Core) redeemMatchGroup(t *trackedTrade, matches []*matchTracker, errs *
 		)
 	}
 
-	// Send the redemption information to the DEX.
+	// Save redemption details and send the redeem message to the DEX.
+	// Saving the redemption details now makes it possible to resend the
+	// `redeem` reqeust at a later time if sending it now fails.
 	for i, match := range matches {
-		err := c.finalizeRedeemAction(t, match, coinIDs[i])
-		if err != nil {
-			errs.addErr(err)
+		proof := &match.MetaData.Proof
+		coinID := []byte(coinIDs[i])
+		if match.Match.Side == order.Taker {
+			match.SetStatus(order.MatchComplete) // could this cause the match to be retired before the `redeem` request succeeds?
+			proof.TakerRedeem = coinID
+		} else {
+			match.SetStatus(order.MakerRedeemed)
+			proof.MakerRedeem = coinID
+		}
+		if err := t.db.UpdateMatch(&match.MetaMatch); err != nil {
+			errs.add("error storing swap details in database for match %s, coin %s: %v",
+				match.id, coinIDString(t.wallets.fromAsset.ID, coinID), err)
 		}
+
+		c.sendRedeemAsync(t, match, coinIDs[i], proof.Secret)
 	}
 }
 
-// finalizeRedeemAction sends a `redeem` request for the specified match,
-// waits for and validates the server's acknowledgement, then saves the
-// redeem details to db.
-// The redeem details are always saved even if sending the `redeem` request
-// errors or a valid ack is not received from the server. This makes it
-// possible to resend the `redeem` request at a later time.
+// sendInitAsync starts a goroutine to send an `redeem` request for the specified
+// match and save the server's ack sig to db. Sends a notification if an error
+// occurs while sending the request or validating the server's response.
 //
 // This method modifies match fields and MUST be called with the trackedTrade
 // mutex lock held for writes.
-func (c *Core) finalizeRedeemAction(t *trackedTrade, match *matchTracker, coinID []byte) error {
-	_, _, proof, auth := match.parts()
-	if len(auth.RedeemSig) != 0 {
-		return fmt.Errorf("'redeem' already sent for match %v", match.id)
```

Comment:
I suppose it's removed because it should now be impossible to have simultaneous calls to `sendRedeemAsync` and with `resendPendingRequests` checking `RedeemSig` too, this does seem superfluous.